### ISSUE TITLE
[Error] ReferenceError: Can't find variable: global

### ISSUE THREAD

**Issue by reyamir** [brillout/vite-plugin-ssr#806]
### Description

Hello,
I'm migrating my next.js app to vps. I set `clientRouting = true` and use `navigate('<url>')` to make redirect to other page. But I got error below

### Error Message + Error Stack

```shell
[Error] Unhandled Promise Rejection: Error: [vite-plugin-ssr@0.4.115][Bug] You stumbled upon a bug in vite-plugin-ssr's source code. Go to https://github.com/brillout/vite-plugin-ssr/issues/new and copy-paste this error. (The error's sta...

	(anonymous function) (useClientRouter.js:191)
```

**Comment by brillout** [comment]
Thanks for the bug report.

What is the content of the line `useClientRouter.js:191`? (The `assert()` line.)

**Comment by reyamir** [comment]
is this?
<img width="898" alt="image" src="https://user-images.githubusercontent.com/123083837/233779341-b7bee53b-a1ab-420a-bf53-2a52c711a7f2.png">

and

<img width="898" alt="image" src="https://user-images.githubusercontent.com/123083837/233779387-78de6e72-f69a-4b34-afdc-fceb19bbd19c.png">


the error message don't show full

**Comment by reyamir** [comment]
my navigate code
```
navigate('/onboarding', { overwriteLastHistoryEntry: true });
```

**Comment by brillout** [comment]
No, the line should contain an `assert()` call. 

Or a reproduction.

**Comment by brillout** [comment]
I think a reproduction would be the best. As a GitHub repository.

**Comment by brillout** [comment]
From first sight I don't see any problems with your code. So, yea, I'm afraid I'll need a reproduction. Feel free to invite me to a private repo.

**Comment by reyamir** [comment]
my app is open source too
you can look at this repo: [https://github.com/luminous-devs/lume/tree/ditch-nextjs](https://github.com/luminous-devs/lume/tree/ditch-nextjs)

to run, just run:

```
pnpm install
```

then

```
pnpm tauri dev
```

**Comment by reyamir** [comment]
oh sorry you, I notify this issue not cause by `navigate()`, when I replace it with `window.location.replace` I still get same error

**Comment by brillout** [comment]
Neat project. Let me have look. It could be because of tauri, let's see.

**Comment by brillout** [comment]
I could reproduce. I'm digging.

**Comment by reyamir** [comment]
oh, I've found the problem, if I add
```
define: {
    global: {
      window: {}
    },
  },
```

to vite config, this issue is gone, but after redirect with `navigate()`, I got hydration error

**Comment by brillout** [comment]
If you fix the original error (the `gloabl` thing) then I believe VPS's error will go away as well.

This means you can continue develop your app while I fix the bug.

**Comment by brillout** [comment]
>  I got hydration error

That's user land though.

That's good: nothing is blocking you anymore, correct?

**Comment by reyamir** [comment]
yes, everything seem fine now, thank you so much

**Comment by brillout** [comment]
👍 Really cool to see a Nostr client built with VPS. Definitely keep me updated.

Also a new official example `/example/tauri/` would be lovely, PR welcome in case that's something you'd be up to.

**Comment by reyamir** [comment]
wow, you know about Nostr too
I'll make a tauri example for VPS later, look like everything work fine now, Nextjs appDir didn't work well with Tauri, I'm tired of wait bug fix

**Comment by brillout** [comment]
Yea...

**Comment by brillout** [comment]
In case you need it, the fix is released in `0.4.115-draft.64315`.
